storage/src/nodestore/persist.rs

```diff
@@ -247,3 +364,207 @@ impl NodeStore<Arc<ImmutableProposal>, FileBacked> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+#[expect(clippy::unwrap_used, clippy::indexing_slicing)]
+mod tests {
+    use super::*;
+    use crate::{
+        Child, HashType, LinearAddress, NodeStore, Path, SharedNode,
+        linear::memory::MemStore,
+        node::{BranchNode, LeafNode, Node},
+        nodestore::MutableProposal,
+    };
+
+    /// Helper to create a test node store with a specific root
+    fn create_test_store_with_root(root: Node) -> NodeStore<MutableProposal, MemStore> {
+        let mem_store = MemStore::new(vec![]).into();
+        let mut store = NodeStore::new_empty_proposal(mem_store);
+        store.mut_root().replace(root);
```

Comment:
nit: why is this function `mut_root()` instead of `root_mut()`?

Reply:
Yeah it should be. I can do that as a different PR.